src/gui/FileBrowser.cpp

```diff
@@ -750,19 +752,19 @@ void FileBrowserTreeWidget::previewFileItem(FileItem* file)
 	// handling() rather than directly creating a SamplePlayHandle
 	if (file->type() == FileItem::FileType::Sample)
 	{
-		TextFloat * tf = TextFloat::displayMessage(
-			tr("Loading sample"),
-			tr("Please wait, loading sample for preview..."),
-			embed::getIconPixmap("sample_file", 24, 24), 0);
-		// TODO: this can be removed once we do this outside the event thread
-		qApp->processEvents(QEventLoop::ExcludeUserInputEvents);
-		if (auto buffer = SampleLoader::createBufferFromFile(fileName))
+		try
 		{
-			auto s = new SamplePlayHandle(new lmms::Sample{std::move(buffer)});
-			s->setDoneMayReturnTrue(false);
-			newPPH = s;
+			newPPH = new SamplePreviewPlayHandle(PathUtil::qStringToPath(fileName));
+		}
+		catch (const std::runtime_error&)
+		{
```

Comment:
Maybe print the error message to the console here?

Reply:
This trickery was because DrumSynth files aren't audio files, so libsndfile can't open them, but I think this is a good use for `sf_open_virtual`.